### PR TITLE
Added String.truncate

### PR DESCRIPTION
**Pull request by CyrusNuevoDia**
Added String.truncate, as per #63

### CODE REVIEW

**Comment by duijf** [comment]
Could you add docs for the options? :)

**Comment by CyrusNuevoDia** [comment]
:shipit: ? :smile:

**Comment by duijf** [comment]
We really need a standard format for docs :P I'll propose something tomorrow/Tuesday.

Final request, could you squash the merge commit?

**Comment by CyrusNuevoDia** [comment]
@duijf I don't know how :laughing: could you send me a tutorial you think is good?

**Comment by knrz** [comment]
Re: function heads defaulting to an empty list, we could, or in `Crutches.Options.combine!` we could compare the provided options with the defaults, and if they're equal, return early. #uselessoptimization?

**Comment by duijf** [comment]
Already sent it over in chat! For others that are curious, [interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Changing-Multiple-Commit-Messages) is the way to go

File: lib/crutches/string.ex [hunk 1]
```diff
@@ -211,4 +212,65 @@ defmodule Crutches.String do
...
+    else
+      string
+    end
+  end
+
+  defp do_truncate(string, length, sep, omission) when is_nil(sep) do
+    String.slice(string, 0, length) <> omission
+  end
+  defp do_truncate(string, length, sep, omission) when is_binary(sep) do
+    {:ok, sep} = Regex.compile(sep)
```
Comment (druzn3k) [inline lib/crutches/string.ex]: What if the separator is a `.`? I think this could break this function.

Comment (CyrusNuevoDia) [inline lib/crutches/string.ex]: Good call. Let me test.

Edit: Fixed. It feels like it's got too many loops (read: `Enum.function`) though.

**Comment by CyrusNuevoDia** [comment]
It now handles string separators as expected :shipit:
Thanks @druzn3k for catching it!

**Comment by duijf** [comment]
The rest looks good :)

**Comment by CyrusNuevoDia** [comment]
Fixed :shipit: :+1:

**Comment by duijf** [comment]
Extra `]` ;)

**Comment by CyrusNuevoDia** [comment]
Doh!

**Pull request merged**